pkg/klusterlet/clusterinfo/loggingInfo_sync.go

```diff
@@ -0,0 +1,254 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	routev1 "github.com/openshift/api/route/v1"
+	openshiftclientset "github.com/openshift/client-go/config/clientset/versioned"
+	routeclient "github.com/openshift/client-go/route/clientset/versioned"
+	clusterv1beta1 "github.com/stolostron/cluster-lifecycle-api/clusterinfo/v1beta1"
+	"github.com/stolostron/multicloud-operators-foundation/pkg/klusterlet/agent"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+type loggingInfoSyncer struct {
+	clusterName    string
+	agentName      string
+	agentNamespace string
+	agentPort      int32
+	agent          *agent.Agent
+	kubeClient     kubernetes.Interface
+	configV1Client openshiftclientset.Interface
+	routeV1Client  routeclient.Interface
+}
+
+func (s *loggingInfoSyncer) sync(ctx context.Context, clusterInfo *clusterv1beta1.ManagedClusterInfo) error {
+	// refresh logging server CA if CA is changed
+	s.refreshAgentServer(clusterInfo)
+
+	clusterInfo.Status.LoggingPort = corev1.EndpointPort{
+		Name:     "https",
+		Protocol: corev1.ProtocolTCP,
+		Port:     s.agentPort,
+	}
+
+	agentEndpoint, err := s.getEndpointAddress(ctx, clusterInfo.Status.KubeVendor)
+	if err != nil {
+		return fmt.Errorf("failed to get logging agent config. error %v", err)
+	}
+	clusterInfo.Status.LoggingEndpoint = *agentEndpoint
+
+	return nil
+}
+
+func (s *loggingInfoSyncer) refreshAgentServer(clusterInfo *clusterv1beta1.ManagedClusterInfo) {
+	select {
+	case s.agent.RunServer <- *clusterInfo:
+		klog.Info("Signal agent server to start")
+	default:
+	}
+
+	s.agent.RefreshServerIfNeeded(clusterInfo)
```

Comment:
in the `RefreshServerIfNeeded` func, it will print the caData in some situation, not sure if we print the caData to log, please check?